### PR TITLE
Add more details cases

### LINKED ISSUE #46: Pass all tests which do not require changes in parsing rules
Depends on PR  #24

### PR DESCRIPTION
**Pull request by dimus**
Fix #46

### CODE REVIEW

File: src/test/resources/test_data.txt [hunk 1]
```diff
@@ -285,20 +284,20 @@ Polyrhachis orsyllus nat musculus Forel 1901|{"scientificName":{"id":"3392132e-3
...
+Glomopsis lonicerae Peck ex C.J. Gould 1945|{"scientificName":{"id":"422687ca-7f4b-5720-8d99-88695f765530","parsed":true,"parser_version":"test_version","verbatim":"Glomopsis lonicerae Peck ex C.J. Gould 1945","normalized":"Glomopsis lonicerae Peck ex C. J. Gould 1945","canonical":"Glomopsis lonicerae","hybrid":false,"virus":false,"details":[{"genus":{"string":"Glomopsis"},"species":{"string":"lonicerae","authorship":"Peck ex C. J. Gould 1945","basionymAuthorTeam":{"author":["Peck"],"year":"1945","exAuthorTeam":{"author":["C. J. Gould"]}}}}]}}
```
Comment (alexander-myltsev) [inline src/test/resources/test_data.txt]: If `"parser_run":1` is not required then it should be removed in all test cases in a separate PR.

File: src/main/scala/org/globalnames/parser/ParserClean.scala [hunk 2]
```diff
@@ -310,8 +310,8 @@ class ParserClean extends SimpleParser {
 
   val unknownAuthor: Rule1[Author] = rule {
     capture("?" |
-            (("auct." | "auct" | "anon." | "anon" | "ht." | "ht" | "hort." |
-              "hort") ~ &(spaceChars | EOI))) ~>
+            (("auct" | "anon" | "ht" | "hort") ~ '.'.? ~
```
Comment (alexander-myltsev) [inline src/main/scala/org/globalnames/parser/ParserClean.scala]: Do we need to capture exact string of anonymous author? In other words, should anonymous author be normalized to, say, `"anon."`, or exact string from verbatim should be preserved?

Comment (alexander-myltsev) [inline src/main/scala/org/globalnames/parser/ParserClean.scala]: I am asking because in Details you simply generate `""unknown"`.

**Comment by alexander-myltsev** [comment]
I made minor improvements. Please, make update on your side.

File: src/test/resources/test_data.txt (hunk 1 above)
Comment (dimus) [inline src/test/resources/test_data.txt]: yes we won't need it anymore

File: src/main/scala/org/globalnames/parser/ParserClean.scala (hunk 2 above)
Comment (dimus) [inline src/main/scala/org/globalnames/parser/ParserClean.scala]: hort. anon. auct. mean different types of unknown, so it is better to keep them in normalized version and in authorship, in semantic explanation I chose to use 'unknown' before and noone complained.

**Comment by dimus** [comment]
all your PRs are included

**Comment by dimus** [comment]
One thing I noticed that by introducing f. instead of all possible forma abbreviations in some rare cases we will introduce ambiguity where like `Aus bus L. forma bus` was before normalilzation and all was clear, but after normalization it is `Aus bus L. f. bus` which can be interpreted as `L. f.` author. I will ask nomenclators what would they recommend

**Comment by dimus** [comment]
Also I think it is good to distinguish different types of `basyonym authorhsip` and add a flag `original_combination` and set it to true where authors are without parenthesis

**Comment by alexander-myltsev** [comment]
> Also I think it is good to distinguish different types of basyonym authorhsip and add a flag original_combination and set it to true where authors are without parenthesis

Make an issue for that, please.

**Comment by alexander-myltsev** [comment]
I cleaned up commits. All is good except JNothing manipulations. But that is a problem of json4s. We would switch to another lib someday.

**Pull request merged**